src/globus_cli/termio/terminal_info.py

```diff
@@ -0,0 +1,64 @@
+"""
+Global interface for managing accessing and modifying terminal info.
+
+Some printers and formatters within this termio module are concerned with text wrapping.
+To do this effectively, they need to know where exactly to start wrapping text.
+"""
+
+from __future__ import annotations
+
+import contextlib
+import shutil
+import typing as t
+from textwrap import TextWrapper
+
+__all__ = ("TERM_INFO", "TerminalTextWrapper")
+
+
+class VirtualTerminalInfo:
+    MIN_COLUMNS = 6
+
+    def __init__(self) -> None:
+        self._column_delta = 0
```

Comment:
May I suggest that we name this to align with `indented()` as the indent? Also requires that we invert some of the arithmetic around it.

```suggestion
        self._current_indent = 0
```

I suggest this after a moment of confusion reading the `+=`, `-=`, etc during review.